### ISSUE TITLE
Handling Varying Length Input Sequences

### ISSUE THREAD

**Issue by ekurtgl** [benedekrozemberczki/pytorch_geometric_temporal#112]
Hi,

I am trying to use pytorch_geometric_temporal framework to make prediction for varying length time-series data, but I couldn't see any example of how varying time dimension is handled in any of the datasets. All of them have fixed time-length. Also how batching  can be handled when we have varying lengths? Using batch size of 1 might be an option but it would slow down the training process. Is there any example code or tutorial for varying length time-series data? Thank you.

Best,
Emre

**Comment by alexriedel1** [comment]
One easy approach for your problem is zero-padding. If it works well or not might depend on your data though

**Comment by ekurtgl** [comment]
It was not really a good option for my case with LSTMs. So, I ended up using varying length sequences with them. I wonder if any of pytorch_geometric_temporal layers can handle varying length sequences. I also reviewed the documentation of this module but I was not able to find any examples with custom datasets. I am kind of new with GNNs and I was able to create my own non-temporal datasets following the [doc](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html), and able to train basic GNN models, but I couldn't see much examples with this module. Can you provide some references to appropriately understand and use this module? Thank you.

**Comment by benedekrozemberczki** [comment]
You could look at the architecture of the data loaders @ekurtgl. Could you please star the repo and cite the design paper if you us it?
